Wrote smolagents blog

Will-Cross1 opened the pull request:
added 3 images
added .DS_Store to gitignore

Will-Cross1 committed:
created smolagents blog, added 3 images

added .DS_Store to gitignore

nmassey001 commented:
Hi @Will-Cross1 Here are my comments:

Smolagents blog

Intro
of setting it up : to setting it up
Along with useful links : Also, I will provide useful links
Transformers : what is a transformer? (e.g. the technique a LLM uses to learn? Inference?)
This guide: this guide at [huggingface.co](http://huggingface.co/) (I like to know where I’m going when I click an external link!)

Setup

Prerequisites
Missing full stops in this section!
What do you mean by “Next open your prefered python interpreter”? (Also sp on preferred)
“Other installs should you need them” : what are these?  Are they definitely needed, or are they just nice to have? 
Available LLMs
some of which are “gated model” : some of which are “gated models” 

Securing Your API Token
inside the json file (we will name api_keys.json) : Missing capital “I” at start of sentence.  Just use : “, which we will name `api_keys.json`.

In the `#login to HuggingFace` code, it’s good that you are trapping the exception, but you then `pass`.  Maybe do something else: raise an exception, exit with an error message, e.g. `raise SystemExit(“Could not login to HuggingFace with access key”)`

No brackets needed around the `However` clause

“we will continue assuming you have done this.” : the examples below will assume that you have done this.

Starting the model with smolagents (API)
For this code to work, do you have to have logged in using the previous bit of code?  Would be worth adding a sentence about this.
You choose your model from Hugging Face’s models and use HfApiModel to utalise it : sp on utilise, maybe use “Hugging Face’s available models”
(Also missing full stops)

It seems models can’t be too large when using the API : add “, the following error can be produced: `too large…`”

What is Ollama - needs a question mark : ?
Ollama is a local runtime: local runtime what?  Library?  Middleware?

Getting Ollama
Missing full stops in this section

on screan : on screen
It will at the end : need commas : i.e, It will, at the end, 
(but I assume that the agent will work with any model) : remove brackets, use a comma. i.e. … example I use, but I will assume that .. 

Figure 2: can you reduce the size of this image?

along with the parameters  … Use a full stop and capital A, rather than a comma.
quantisation level is which is a measure of how compressed the LLM is : comma before which
Use full stop and new sentence rather than brackets.

I setteled : sp 
Remove so, use full stop and capital Y
“(If it says Ollama command not found” - Remove brackets.  Put ollama command not found in `` for code.
“(but there is more”. Remove brackets, start new sentence, i.e. There is more information
readong : sp (reading)

Starting the model with smolagents (Ollama)

Before running the code, do you need to log-in with hugging face?
Full stops!
num_ctx - put in `` for code
huggingface - capitals?
to calculate - needs capital

Performance Notes

e.g: I ran the same python file multiple times and timed it - Start new sentence: For example, I ran the same … “it would go through each going through each step in a round robin approach that took about the same length of time as it would each ran one at a time I am sure there is a way to improve it but that is out of the scope of this blog” : rewrite this chunk.  Capital to start.
intense : intensive
windows : capital, also maybe for task manager
Full stops! Can you reduce the size of the “memory pressure figure”?
“The computer I used didn’t have a graphics card” - The Mac M1s do have GPUs, but they’re not very powerful compared to a Nvidia A100! :)
 Aditional notes - sp (Additional)

Agents and Their Applications

Full stops!  Inconsistent use of capitals for smolagents and Hugging Face
“read the [agents documentation](https://huggingface.co/docs/smolagents/v1.6.0/en/reference/agents) for smolagents. And this guide provided by Hugging Face is helpful for [building good agents](https://huggingface.co/docs/smolagents/tutorials/building_good_agents)” : To understand the differences between the agents, read the [agents documentation](https://huggingface.co/docs/smolagents/v1.6.0/en/reference/agents) for smolagents, and this guide provided by Hugging Face is helpful for [building good agents](https://huggingface.co/docs/smolagents/tutorials/building_good_agents).

MultiStepAgent
Full stops! Table is not formatted correctly

CodeAgent
aditional - sp
Remove brackets, just use a comma

ToolCallingAgent
Remove brackets.  Sentence is fine on its own
Full stops.

manager_agent to run use it - I think you mean CodeAgent, as `manager_agent` is just the instance of the CodeAgent class.  Use `` around things that are referenced in the code to put them in the same font as the code.  “Run use” - either of these will do! :)

aditional - sp again
More info - expand to something like “more information can be found in the ….”

Example: Decorator
model = … : what does this refer to?  Put a comment in the code (i.e. using #) with a comment about where the model comes from

Example: subclass
Same comment as above.  Where does mode come from?

Which to chose - add a question mark?
Here is a table that can help you choose whether to use a subclass or decorator: Table not formatted correctly

Simple example
“The too is” : The tool is

Weak LLM Performance, Solutions: 1. Ensure the model is powerful enough: maybe put a sentence in here about how you can determine the power of the model. e.g. the number of parameters / it’s “B” number
3. num_ctx in `` to make code formatted
Full stops

Not enough steps - Cause
the default - needs capital T

Solutions
Need capitals in list to be consistent

Poor description - Cause
Inproperly - Improperly 

Solutions
2. detaile - sp - detail
3. Add more type hints - maybe link to Python typing module

Look here and … here on the Hugging Face website.

More complex example
Could add a lead in sentence, such as “To summarise, in this section I’ll walk ….”

My thoughts and findings
 setup - set up
“I recommend using a local LLM” - as shown in section …

Sources and further information
These need to be links! :)

Will-Cross1 committed:
Made suggested changes from Neil

Wasn't sure how to decrease image size, I tried to but I can't check that it worked without Jekyll.

I'm not sure what you mean about the tables being formatted incorrectly, they look fine in the preview?

nmassey001 committed:
Proof read and corrected Will's Smolagents blog post

nmassey001 reviewed (approved):
I have proof read and made a few corrections.
All good and ready for publication!
